services/wfm/src/main/java/org/openkilda/wfm/topology/flow/service/FlowService.java

```diff
@@ -377,20 +428,25 @@ private void updateIslAvailableBandwidth(SwitchId srcSwitchId, int srcPort, Swit
                 });
     }
 
-    private List<FlowSegment> deleteFlowSegments(Flow flow) {
-        List<FlowSegment> segments =
-                Lists.newArrayList(flowSegmentRepository.findByFlowIdAndCookie(flow.getFlowId(), flow.getCookie()));
+    private List<FlowSegment> getFlowSegments(Flow flow) {
+        return Lists.newArrayList(flowSegmentRepository.findByFlowIdAndCookie(flow.getFlowId(), flow.getCookie()));
+    }
 
-        segments.forEach(segment -> {
+    private void deleteFlowSegments(List<FlowSegment> flowSegments) {
+        flowSegments.forEach(segment -> {
             log.debug("Deleting the flow segment: {}", segment);
 
             flowSegmentRepository.delete(segment);
 
             updateIslAvailableBandwidth(segment.getSrcSwitch().getSwitchId(), segment.getSrcPort(),
                     segment.getDestSwitch().getSwitchId(), segment.getDestPort());
         });
+    }
 
-        return segments;
+    private void lockSwitches(List<FlowSegment> flowSegments) {
```

Comment:
Can we put this method in FlowSegmentRepository as lockUsedSwitches(List<FlowSegment> flowSegments) instead of de-encapsulating lockSwitches(Switch... switches)?

Reply:
Here we have an assumption that the flow destination switch will be fetched and locked via the reverse flow segments. This is because of ```switches.add(flowSegment.getSrcSwitch())```. Thus I don't like to move such logic into a repository. Let it be in a service.